### DISCUSSION TITLE
Using @use/@forward over @import for Sass?

### DISCUSSION THREAD

**Discussion by ProspectPyxis** [picocss/pico#203] [General]
Sass has plans to [gradually phrase out `@import` directives in the future](https://github.com/sass/sass/blob/main/accepted/module-system.md#timeline) (ignore the dates on the document), and the recommendation is to use the new `@use` and `@forward` directives instead.

Does picocss have any plans to switch to this new syntax in the future? I'm phrasing this more as a question, since I'm aware this would be a breaking change and would likely require a 2.0 version of the library at the very least.

**Comment by lucaslarroche** [comment]
Hi @ProspectPyxis,
I had never looked at `@use` and `@forward`.

The change is easy to make. I created a test branch here: [`ll/sass/import`](https://github.com/picocss/pico/tree/ll/sass/import).

There are no breaking changes on .css files.
But yes, it would cause breaking changes for all users who use Pico in this way, as recommended in [Customization](https://picocss.com/docs/customization.html):

```SCSS
// Override default variables
$primary-500: #e91e63;
$primary-600: #d81b60;
$primary-700: #c2185b;

// Import full Pico source code
@import "path/pico";
```

Not sure what to do here.
It would be safer to wait for version 2.

Note: v2 will also add a configurable prefix on all CSS variables:

```SCSS
$prefix: "pico-";
:root {
  --#{$prefix}color: #000; // --pico-color: #000;
}
```

**Reply by ProspectPyxis** [reply]
Ah, neat! Yes, my worry was about people already using the customization syntax in the docs. I agree that we should probably wait for v2 before making the move.

The configurable prefix also looks very interesting - looking forward to when this happens!
